src/main/java/pl/edu/pw/mini/ingreedio/api/service/ProductService.java

```diff
@@ -28,8 +34,12 @@ public List<ProductDto> getAllProducts() {
                 .map(productDtoMapper).collect(Collectors.toList());
     }
 
-    public Product getProductById(Long id) {
-        return productRepository.findById(id).orElse(null);
+    public FullProductDto getProductById(Long id) {
+        Product product = productRepository.findById(id).orElse(null);
+        if (product != null) {
+            return fullProductDtoMapper.apply(product);
+        }
+        return null;
```

Comment:
As `findById` returns optional, it would be better to write this like this
```suggestion
        return productRepository.findById(id).map(fullProductDtoMapper::apply).orElse(null);
```